### PR TITLE
store/tikv: move transaction options out to /kv

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
Now that options are not used by `store/tikv`, we can move them out again.

Part of https://github.com/pingcap/tidb/issues/24302

### What is changed and how it works?
What's Changed:
- Move options to kv
- update usages

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

**Review by AndreMouche — APPROVED** [review]
LGTM

**Review by sticnarf — APPROVED** [review]

File: kv/option.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	// IsolationLevel sets isolation level for current transaction. The default level is SI.
+	IsolationLevel
+	// Priority marks the priority of this transaction.
+	Priority
+	// NotFillCache makes this request do not touch the LRU cache of the underlying storage.
+	NotFillCache
+	// SyncLog decides whether the WAL(write-ahead log) of this request should be synchronized.
+	SyncLog
+	// KeyOnly retrieve only keys, it can be used in scan now.
+	KeyOnly
```
Comment (tisonkun) [inline kv/option.go:31]: I think `KeyOnly` is already unused.

Comment (tisonkun) [reply kv/option.go:31]: I mean this variable.

**Pull request merged**